test/yield.js

```diff
@@ -0,0 +1,169 @@
+const should = require("should");
+
+const { ResolverFactory } = require("../");
+const { Volume } = require("memfs");
+
+/** @typedef {import("../lib/Resolver").ResolveContext} ResolveContext */
+/** @typedef {ResolveContext & Required<Pick<ResolveContext, 'yield' | 'fileDependencies' | 'contextDependencies' | 'missingDependencies'>>} StrictResolveContext */
+
+const fileSystem = Volume.fromJSON(
+	{
+		"/a/foo/a": "",
+		"/a/foo/b": "",
+		"/a/foo-2/b": "",
+		"/a/foo-2/c": "",
+		"/b/foo/a": ""
+	},
+	"/"
+);
+
+describe("should resolve all aliases", () => {
+	const resolver = ResolverFactory.createResolver({
+		extensions: [".js"],
+		alias: {
+			index: ["/a/foo", "/a/foo-2"]
+		},
+		modules: "/",
+		fileSystem
+	});
+	const modulesResolver = ResolverFactory.createResolver({
+		extensions: [".js"],
+		modules: ["a", "b"],
+		fileSystem
+	});
+
+	it("should yield all b files", done => {
+		const paths = [];
+		const yield_ = ({ path }) => paths.push(path);
+		const fileDependencies = new Set();
+		const contextDependencies = new Set();
+		const missingDependencies = new Set();
+		/** @type {ResolveContext} */
+		const context = {
+			yield: yield_,
+			fileDependencies,
+			contextDependencies,
+			missingDependencies
+		};
+
+		resolver.resolve({}, "/", "index/b", context, (err, result) => {
+			should(err).be.eql(null);
+			should(result).be.eql(undefined);
+			should(paths).be.eql(["/a/foo/b", "/a/foo-2/b"]);
+			should(Array.from(fileDependencies.values()).sort()).be.eql([
+				"/",
+				"/a",
+				"/a/foo",
+				"/a/foo-2",
+				"/a/foo-2/b",
+				"/a/foo/b"
+			]);
+			should(Array.from(missingDependencies.values()).sort()).be.eql([
+				"/a/foo-2/b",
```

Comment:
Interesting that this is in `missingDependencies` and in `fileDependencies`.
I guess it tries "as file" first and "as directory" second, resulting in it exists as file but is missing as directory...
Doesn't seem to be directly caused by this PR, so probably not relavant.